modin/pandas/base.py

```diff
@@ -333,6 +334,45 @@ def _validate_other(
                 )
         return result
 
+    def _validate_function(self, func, on_invalid=TypeError):
+        """
+        Check the validity of the function which is intended to be applied to the frame.
+
+        Parameters
+        ----------
+        func : object
+        on_invalid : callable(str), default: TypeError
+            Function to call in case invalid `func` is met, `on_invalid` takes an error
+            message as a single argument. If `on_invalid` has an Exception type then it
+            will arise. Raise ``TypeError`` by default.
+            **Note:** This parameter is a hack to concord with pandas error types.
+        """
+
+        def error_raiser(msg, exception=Exception):
+            raise exception(msg)
+
+        if issubclass(on_invalid, Exception):
+            on_invalid = functools.partial(error_raiser, exception=on_invalid)
```

Comment:
@dchigarev , why do we need this?

Reply:
1. We can't just use exception type as a function to throw an exception, there has to be explicit `raise`, so if `on_invalid` is some exception we have to wrap it into a function that will raise it.
2. Since `on_invalid` interface is `callable(str)` but for `error_raiser` it's `callable(str, Exception)` we need to bind a proper exception type to match `error_raiser` to the `on_invalid` contract

Reply:
I see, make sense.